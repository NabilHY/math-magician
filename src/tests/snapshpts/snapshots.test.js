import renderer from 'react-test-renderer';
import Calculator from '../../component/calculator';
import Home from '../../component/home';
import Quote from '../../component/quote';

describe('Calculator component renders correctly', () => {
  it('calculator page matches snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('home component renders correctely', () => {
  it('main page matches snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Quote component renders correctely', () => {
  it('Quote page matches snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

