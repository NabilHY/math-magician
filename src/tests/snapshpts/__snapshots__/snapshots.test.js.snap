// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly calculator page matches snapshot 1`] = `
<section
  className="component"
>
  <h3>
    Let's Do some Math
  </h3>
  <div
    className="calc"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="btn"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn methods"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn methods"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn methods"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn methods"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn span-2"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn methods"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`Quote component renders correctely Quote page matches snapshot 1`] = `
<h2>
  Mathematics is not about numbers, equations, computaions, or algorithms: it is about understanding. William Paul Thurston
</h2>
`;

exports[`home component renders correctely main page matches snapshot 1`] = `
<section>
  <h2>
    Welcome to The Math Magicien
  </h2>
  <p>
     Math magicien was built using React library
  </p>
  <p>
    Feel free to navigate throught the section on this website.
  </p>
  <p>
    Click on the Calculator above to open an application that does simple Math operations for you
  </p>
  <p>
    Click on the quote to generate a random math related quote by a famous mathmaticien
  </p>
</section>
`;
